rosidl_runtime_py/rosidl_runtime_py/set_message.py

```diff
@@ -53,4 +58,9 @@ def set_message_fields(msg: Any, values: Dict[str, str]) -> None:
                     submsg = field_elem_type()
                     set_message_fields(submsg, value[n])
                     value[n] = submsg
+            if isinstance(rosidl_type.value_type, BasicType):
+                field_elem_type = rosidl_type.value_type.typename
+                field_elem_type = eval(SPECIAL_NESTED_BASIC_TYPES[field_elem_type]['dtype'])
+                for n in range(len(field_value)):
+                    value[n] = field_elem_type(field_value[n])
```

Comment:
This last part I added because I thought it could help in the future with some custom or new message array type.

Reply:
If you want to still add this part please update the PR to target `master` as well as describe how this part is being used with reproducible steps.

Reply:
Thanks for the time. I'll close the PR since I believe this extra piece of code won't be of any use now.